### PR TITLE
DOC: change all non-code instances of Numpy to NumPy

### PR DESCRIPTION
**Pull request by pdebuyl**
Instances remain for NumpyVersion and Numpy.rec.fromarrays that are
references to code.

see issue #7986

### CODE REVIEW

**Comment by btel** [comment]
It looks good. However, I would not change the release notes - it might look strange that the last commit is after the release date.

**Comment by pdebuyl** [comment]
Forced pushed the PR with no changes to release notes.

**Comment by btel** [comment]
I also opened an associated PR for the website.

**Comment by pdebuyl** [comment]
Should I also do the code part? The docstrings are full of both both versions, including C files in multiarray, umath and mtrand.

**Comment by btel** [comment]
I think the docstrings and comment could be updated, but not the source code.

**Comment by btel** [comment]
It looks good to me.

File: doc/source/reference/swig.interface-file.rst [hunk 1]
```diff
@@ -1,4 +1,4 @@
-Numpy.i: a SWIG Interface File for NumPy
+NumPy.i: a SWIG Interface File for NumPy
```
Comment (jaimefrio) [inline doc/source/reference/swig.interface-file.rst]: This should be changed to all lower case, numpy.i, which is the actual name of the file it refers to.

**Comment by jaimefrio** [comment]
Thanks a lot for doing this!

That was a lot of NumPy to read, but I found one instance that needs to be corrected, see inline comment. My diagonal looking at the rest didn't find anything else. Should I merge it once you change the SWIG file name, or is there anything else you want to include in this PR?

**Comment by pdebuyl** [comment]
You are welcome :-) It is my pleasure to contribute to NumPy in the extent of my abilities and time!

The PR is complete as it is. There is also a need for a statement about the capitalization somewhere (see #7986) but that is for later.

**Comment by btel** [comment]
@pdebuyl @jaimefrio could you also take a look at numpy/numpy.org#5?

Should these changes be mentioned in the release notes for the next release?

File: doc/source/reference/routines.numarray.rst [hunk 2]
```diff
@@ -2,4 +2,4 @@
 Numarray compatibility
 **********************
 
-The numarray module was removed in Numpy 1.9.
+The numarray module was removed in NumPy 1.9.
```
Comment (charris) [inline doc/source/reference/routines.numarray.rst]: Should make the `1.9.` `1.9.0.` as below.

File: numpy/add_newdocs.py [hunk 3]
```diff
@@ -2014,7 +2014,7 @@ def luf(lamdaexpr, *args, **kwargs):
       were elements.
 
     .. warning::
-       This function is preliminary and included in Numpy 1.10 for testing
+       This function is preliminary and included in NumPy 1.10 for testing
```
Comment (charris) [inline numpy/add_newdocs.py]: `1.10.` -> `1.10.0`

File: numpy/core/code_generators/cversions.txt [hunk 4]
```diff
@@ -22,7 +22,7 @@
 
 # Version 9 (NumPy 1.8) Added interface for partition functions,
 # PyArray_NEW_ZEROED, commented out as the hash changed in
-# Numpy 1.9 due to annotation.
+# NumPy 1.9 due to annotation.
```
Comment (charris) [inline numpy/core/code_generators/cversions.txt]: `1.9.0`

File: numpy/core/numeric.py [hunk 5]
```diff
@@ -759,7 +759,7 @@ def isfortran(a):
 
     This function is obsolete and, because of changes due to relaxed stride
     checking, its return value for the same array may differ for versions
-    of Numpy >= 1.10 and previous versions. If you only want to check if an
+    of NumPy >= 1.10 and previous versions. If you only want to check if an
```
Comment (charris) [inline numpy/core/numeric.py]: `1,10.0`

File: numpy/core/numeric.py [hunk 6]
```diff
@@ -1158,22 +1158,22 @@ def alterdot():
...
+    When NumPy is built with an accelerated BLAS like ATLAS, these
     functions are replaced to make use of the faster implementations.  The
     faster implementations only affect float32, float64, complex64, and
     complex128 arrays. Furthermore, the BLAS API only includes
     matrix-matrix, matrix-vector, and vector-vector products. Products of
     arrays with larger dimensionalities use the built in functions and are
     not accelerated.
 
-    .. note:: Deprecated in Numpy 1.10
+    .. note:: Deprecated in NumPy 1.10
```
Comment (charris) [inline numpy/core/numeric.py]: `1.10.0`

File: numpy/core/numeric.py [hunk 7]
```diff
@@ -1195,10 +1195,10 @@ def restoredot():
     an accelerated BLAS, or when being very careful about benchmarking
     linear algebra operations.
 
-    .. note:: Deprecated in Numpy 1.10
+    .. note:: Deprecated in NumPy 1.10
```
Comment (charris) [inline numpy/core/numeric.py]: `1.10.0`

File: numpy/lib/function_base.py [hunk 8]
```diff
@@ -489,8 +489,8 @@ def histogram(a, bins=10, range=None, normed=False, weights=None,
         based on the actual data within `range`, the bin count will fill
         the entire range including portions containing no data.
     normed : bool, optional
-        This keyword is deprecated in Numpy 1.6 due to confusing/buggy
-        behavior. It will be removed in Numpy 2.0. Use the ``density``
+        This keyword is deprecated in NumPy 1.6 due to confusing/buggy
```
Comment (charris) [inline numpy/lib/function_base.py]: `1.6.0`, `2.0.0`

File: numpy/lib/function_base.py [hunk 9]
```diff
@@ -809,7 +809,7 @@ def histogram(a, bins=10, range=None, normed=False, weights=None,
         else:
             return n, bins
     else:
-        # deprecated, buggy behavior. Remove for Numpy 2.0
+        # deprecated, buggy behavior. Remove for NumPy 2.0
```
Comment (charris) [inline numpy/lib/function_base.py]: See above.

File: numpy/lib/nanfunctions.py [hunk 10]
```diff
@@ -453,7 +453,7 @@ def nansum(a, axis=None, dtype=None, out=None, keepdims=np._NoValue):
     Return the sum of array elements over a given axis treating Not a
     Numbers (NaNs) as zero.
 
-    In Numpy versions <= 1.8 Nan is returned for slices that are all-NaN or
+    In NumPy versions <= 1.8 Nan is returned for slices that are all-NaN or
```
Comment (charris) [inline numpy/lib/nanfunctions.py]: `1.8.0`

File: numpy/linalg/linalg.py [hunk 11]
```diff
@@ -677,7 +677,7 @@ def qr(a, mode='reduced'):
     `a` is of type `matrix`, all the return values will be matrices too.
 
     New 'reduced', 'complete', and 'raw' options for mode were added in
-    Numpy 1.8 and the old option 'full' was made an alias of 'reduced'.  In
+    NumPy 1.8 and the old option 'full' was made an alias of 'reduced'.  In
```
Comment (charris) [inline numpy/linalg/linalg.py]: `1.8.0`

File: numpy/polynomial/_polybase.py [hunk 12]
```diff
@@ -742,7 +742,7 @@ def fit(cls, x, y, deg, domain=None, rcond=None, full=False, w=None,
         deg : int or 1-D array_like
             Degree(s) of the fitting polynomials. If `deg` is a single integer
             all terms up to and including the `deg`'th term are included in the
-            fit. For Numpy versions >= 1.11 a list of integers specifying the
+            fit. For NumPy versions >= 1.11 a list of integers specifying the
```
Comment (charris) [inline numpy/polynomial/_polybase.py]: `1.11.0`

File: numpy/polynomial/chebyshev.py [hunk 13]
```diff
@@ -1620,7 +1620,7 @@ def chebfit(x, y, deg, rcond=None, full=False, w=None):
     deg : int or 1-D array_like
         Degree(s) of the fitting polynomials. If `deg` is a single integer
         all terms up to and including the `deg`'th term are included in the
-        fit. For Numpy versions >= 1.11 a list of integers specifying the
+        fit. For NumPy versions >= 1.11 a list of integers specifying the
```
Comment (charris) [inline numpy/polynomial/chebyshev.py]: `1.11.0`.

File: numpy/polynomial/hermite_e.py [hunk 14]
```diff
@@ -1388,7 +1388,7 @@ def hermefit(x, y, deg, rcond=None, full=False, w=None):
     deg : int or 1-D array_like
         Degree(s) of the fitting polynomials. If `deg` is a single integer
         all terms up to and including the `deg`'th term are included in the
-        fit. For Numpy versions >= 1.11 a list of integers specifying the
+        fit. For NumPy versions >= 1.11 a list of integers specifying the
```
Comment (charris) [inline numpy/polynomial/hermite_e.py]: `1.11.0`

File: numpy/polynomial/laguerre.py [hunk 15]
```diff
@@ -1390,7 +1390,7 @@ def lagfit(x, y, deg, rcond=None, full=False, w=None):
     deg : int or 1-D array_like
         Degree(s) of the fitting polynomials. If `deg` is a single integer
         all terms up to and including the `deg`'th term are included in the
-        fit. For Numpy versions >= 1.11 a list of integers specifying the
+        fit. For NumPy versions >= 1.11 a list of integers specifying the
```
Comment (charris) [inline numpy/polynomial/laguerre.py]: `1.11.0`

File: numpy/polynomial/legendre.py [hunk 16]
```diff
@@ -1421,7 +1421,7 @@ def legfit(x, y, deg, rcond=None, full=False, w=None):
     deg : int or 1-D array_like
         Degree(s) of the fitting polynomials. If `deg` is a single integer
         all terms up to and including the `deg`'th term are included in the
-        fit. For Numpy versions >= 1.11 a list of integers specifying the
+        fit. For NumPy versions >= 1.11 a list of integers specifying the
```
Comment (charris) [inline numpy/polynomial/legendre.py]: `1.11.0`

File: numpy/polynomial/polynomial.py [hunk 17]
```diff
@@ -1309,7 +1309,7 @@ def polyfit(x, y, deg, rcond=None, full=False, w=None):
     deg : int or 1-D array_like
         Degree(s) of the fitting polynomials. If `deg` is a single integer
         all terms up to and including the `deg`'th term are included in the
-        fit. For Numpy versions >= 1.11 a list of integers specifying the
+        fit. For NumPy versions >= 1.11 a list of integers specifying the
```
Comment (charris) [inline numpy/polynomial/polynomial.py]: `1.11.0`

File: numpy/random/mtrand/mtrand.pyx [hunk 18]
```diff
@@ -2547,7 +2547,7 @@ cdef class RandomState:
         Parameters
         ----------
         df : int or array_like of ints
-            Degrees of freedom, should be > 0 as of Numpy 1.10,
+            Degrees of freedom, should be > 0 as of NumPy 1.10,
```
Comment (charris) [inline numpy/random/mtrand/mtrand.pyx]: `1.10.0`.

**Comment by charris** [comment]
I made comments about changing version references from the form `1.10.` to `1.10.0`. Strictly speaking, that is unrelated to the NumPy changes, but I think it best to settle on using a full version string.

**Comment by pdebuyl** [comment]
Hi,
I made the changes and spotted a few more instances close to existing capitalization changes, that I also updated.

**Comment by jaimefrio** [comment]
Thanks Pierre!

**Comment by pdebuyl** [comment]
Thanks @btel @charris @jaimefrio for guidance :-)

**Pull request merged**